Test/hofs/Examples.dfy

```diff
@@ -1,57 +0,0 @@
-// RUN: %testDafnyForEachCompiler "%s" --refresh-exit-code=0 -- --relax-definite-assignment
-
-function Apply<A,B>(f: A ~> B, x: A): B
-  reads f.reads(x)
```

Comment:
Why is this example not supported? It's no longer sound?

Reply:
Correct. If you want to call method f.reads, you need to know that its precondition holds. It used to be that the axioms were incorrect and that you could call that function without its precondition being met, but not anymore.

Reply:
It seems as if 
```
function Apply'<A,B>(f: A ~> B) : A ~> B
{
  x requires f.requires(x)
    reads assume f.requires(x); f.reads(x)
    => f(x)
}
```
verifies, but 
```
function Apply'<A,B>(f: A ~> B) : A ~> B
{
  x requires f.requires(x)
    reads assert f.requires(x); f.reads(x)
    => f(x)
}
```
fails. Shouldn't I be able to deduce the assertions from the precondition?

Reply:
Perhaps, but let's not confuse distinct problems. The purpose of this PR is to ensure that when a .reads function is called, its precondition is checked accordingly. This is a matter of soundness. It is not a goal of this PR to solve incompleteness problems with higher-order functions, including ones that result from now using sound axioms. We should instead file an issue when we believe that something ought to be verified and doesn't.